data/src/main/java/org/apache/iceberg/MetricsAppender.java

```diff
@@ -0,0 +1,839 @@
+    }
+
+    // TODO or setRecordRepresentation(RecordRepresentation) for every option?
+    public Builder useRecord() {
+      recordRepresentation = RecordRepresentation.RECORD;
+      return this;
+    }
+
+    public Builder useGenericRecord() {
+      recordRepresentation = RecordRepresentation.GENERIC_RECORD;
+      return this;
+    }
+
+    public Builder useByteArrayForFixedType() {
+      fixedTypeRepresentation = FixedTypeRepresentation.BYTE_ARRAY;
+      return this;
+    }
+
+    public Builder useByteBufferForFixedType() {
+      fixedTypeRepresentation = FixedTypeRepresentation.BYTE_BUFFER;
+      return this;
+    }
+
+    public Builder useGenericFixedForFixedType() {
+      fixedTypeRepresentation = FixedTypeRepresentation.GENERIC_FIXED;
+      return this;
+    }
+
+    public Builder useIntForDateType() {
+      dateTypeRepresentation = DateTypeRepresentation.INT;
+      return this;
+    }
+
+    public Builder useLocalDateForDateType() {
+      dateTypeRepresentation = DateTypeRepresentation.LOCAL_DATE;
+      return this;
+    }
+
+    public Builder useLongForTimeType() {
+      timeTypeRepresentation = TimeTypeRepresentation.LONG;
+      return this;
+    }
+
+    public Builder useLocalTimeForTimeType() {
+      timeTypeRepresentation = TimeTypeRepresentation.LOCAL_TIME;
+      return this;
+    }
+
+    public Builder useLongForTimestampType() {
+      timestampTypeRepresentation = TimestampTypeRepresentation.LONG;
+      return this;
+    }
+
+    public Builder useDateTimeForTimestampType() {
+      timestampTypeRepresentation = TimestampTypeRepresentation.OFFSET_DATETIME;
+      return this;
+    }
+
+    public Builder useLocalDateTimeForTimestampType() {
+      timestampTypeRepresentation = TimestampTypeRepresentation.LOCAL_DATETIME;
+      return this;
+    }
+
+
+    public MetricsAppender<D> build() {
+      // TODO check iif not defaults??
+      return new MetricsAppender<>(appender, schema,
+          new MetricsAppenderConfiguration(recordRepresentation,
+              fixedTypeRepresentation,
+              dateTypeRepresentation,
+              timeTypeRepresentation,
+              timestampTypeRepresentation));
+    }
+  }
+
+  private static class MetricsCollectorMap<K, V> extends MetricsCollectorBase<Map<K, V>> {
+    private final MetricsCollector collectorKey;
+    private final MetricsCollector collectorValue;
+
+    MetricsCollectorMap(MetricsCollector<K> collectorKey, MetricsCollector<V> collectorValue) {
+      this.collectorKey = collectorKey;
+      this.collectorValue = collectorValue;
+    }
+
+    @Override
+    public void add(Map<K, V> map) {
+      if (map != null) {
+        for (K k : map.keySet()) {
+          collectorKey.add(k);
+          collectorValue.add(map.get(k));
+        }
+      } else {
+        collectorKey.add(null);
+        collectorValue.add(null);
+      }
+    }
+
+    @Override
+    public Stream<FieldMetrics> getMetrics() {
```

Comment:
Minor: we generally leave off the `get` because it doesn't add value to method names.